### PR TITLE
Added feature to parse and detect using tika on complete folder recursively

### PR DESCRIPTION
**Pull request by yashtanna93**

### CODE REVIEW

File: tika/tika.py [hunk 1]
```diff
@@ -95,13 +94,13 @@ def runCommand(cmd, option, urlOrPaths, port, outDir=None, serverHost=ServerHost
...
     serverEndpoint = 'http://' + serverHost + ':' + port
     if cmd == 'parse':
-        if len(urlOrPaths) == 1:
-            status, resp = parse1(option, urlOrPaths[0], serverEndpoint, verbose, tikaServerJar)
-	    if encode:
-		resp = resp.encode("utf-8")
-            return resp
-        else:
-            return parseAndSave(option, urlOrPaths, outDir, serverEndpoint, verbose, tikaServerJar)
+#        if len(urlOrPaths) == 1:
```
Comment (chrismattmann) [inline tika/tika.py]: if it's commented out, can you please remove from the PR? @yashtanna93

File: tika/tika.py [hunk 2]
```diff
@@ -95,13 +94,13 @@ def runCommand(cmd, option, urlOrPaths, port, outDir=None, serverHost=ServerHost
...
+#            status, resp = parse1(option, urlOrPaths[0], serverEndpoint, verbose, tikaServerJar)
+#            if encode:
+#                resp = resp.encode("utf-8")
+#            return resp
+#        else:
```
Comment (chrismattmann) [inline tika/tika.py]: commented please remove

**Comment by chrismattmann** [comment]
@yashtanna93 please review my comments and update.

**Comment by yashtanna93** [comment]
Hi Prof. Mattmann,
I will update it tonight.

**Comment by yashtanna93** [comment]
Removed the comments

File: tika/tika.py [hunk 3]
```diff
@@ -86,7 +86,6 @@
 def echo2(*s): sys.stderr.write('tika.py: ' + ' '.join(map(str, s)) + '\n')
 def warn(*s):  echo2('Warn:', *s)
 def die(*s):   warn('Error:',  *s); echo2(USAGE); sys.exit()
-def setTranslator(translator): Translator = translator
```
Comment (chrismattmann) [inline tika/tika.py]: why did you remove this?

File: tika/tika.py [hunk 4]
```diff
@@ -68,8 +68,8 @@
 import hashlib
 import platform
 from subprocess import Popen
-from subprocess import PIPE
```
Comment (chrismattmann) [inline tika/tika.py]: why did you remove this?

File: tika/tika.py [hunk 5]
```diff
@@ -121,7 +114,15 @@ def parseAndSave(option, urlOrPaths, outDir=None, serverEndpoint=ServerEndpoint,
     """Parse the objects and write extracted metadata and/or text in JSON format to matching
     filename with an extension of '_meta.json'."""
     metaPaths = []
-    for path in urlOrPaths:
+    paths = []
```
Comment (chrismattmann) [inline tika/tika.py]: we should refactor this into a function since you use it in a lot of places

File: tika/tika.py [hunk 6]
```diff
@@ -395,7 +419,6 @@ def main(argv=None):
         die("%s error: Bad option: %s, %s" % (argv[0], bad_opt, msg))
         
     tikaServerJar = TikaServerJar
-    serverEndpoint = ServerEndpoint
```
Comment (chrismattmann) [inline tika/tika.py]: why was this removed?

File: tika/tika.py [hunk 7]
```diff
@@ -273,7 +298,6 @@ def checkTikaServer(serverHost=ServerHost, port = Port, tikaServerJar=TikaServer
     urlp = urlparse(tikaServerJar)
     serverEndpoint = 'http://' + serverHost +':' + str(port)
     jarPath = os.path.join(TikaJarPath, 'tika-server.jar')
-    logPath = os.path.join(TikaJarPath, 'tika-server.log')
```
Comment (chrismattmann) [inline tika/tika.py]: why was this removed?

Comment (yashtanna93) [inline tika/tika.py]: It is not been used anywhere

File: tika/tika.py (hunk 6 above)
Comment (yashtanna93) [inline tika/tika.py]: It is not been used anywhere

File: tika/tika.py (hunk 4 above)
Comment (yashtanna93) [inline tika/tika.py]: It was not in use

File: tika/tika.py (hunk 3 above)
Comment (yashtanna93) [inline tika/tika.py]: It was not in use

**Comment by yashtanna93** [comment]
Code refactored to use function

**Comment by chrismattmann** [comment]
looks good, merging thanks @yashtanna93 !
